src/main/webapp/babel-src/feedBackBox.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import Dialog from 'material-ui/Dialog';
+import FlatButton from 'material-ui/FlatButton';
+import TextField from 'material-ui/TextField';
+import Snackbar from 'material-ui/Snackbar';
+import {FEEDBACK_CHAR_LIMIT, FEEDBACK_SNACKBAR_AUTO_HIDE_DURATION, UI_STRINGS, INLINE_STYLES} from './util';
+
+export class FeedBackBox extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            charCtr : 0,
+            errorMsg : '',
+            feedbackMsg: '',
+            sendSuccess : false,
+            sendError : false
+        };
+
+        // this.updateTextField = this.updateTextField.bind(this);
+    }
+
+    updateTextField(event,value) {
+
+        if (value.length < FEEDBACK_CHAR_LIMIT) {
+            this.setState({
+               charCtr : value.length,
+               errorMsg : '',
+               feedbackMsg : value
+            });
+        } else {
+            this.setState({
+                charCtr : FEEDBACK_CHAR_LIMIT,
+                errorMsg : UI_STRINGS.FEEDBACK_CHAR_LIMIT_ERROR_MSG
+            })
+        }
+    }
+
+    sendFeedBackMsg() {
+        $.ajax({
+            type: "POST",
+            url: "api/feedback",
+            data: JSON.stringify({"message": this.state.feedbackMsg}),
+            success: (response) => {
+                var responseObject = JSON.parse(response);
+                if (responseObject.success === true) {
+                    this.setState({
+                        sendSuccess : true,
+                        sendError : false
+                    });
+                } else {
+                    this.setState({
+                        sendSuccess : false,
+                        sendError : true
+                    })
+                }
+            },
+            error: (response) => {
+                this.setState({
+                    sendSuccess: false,
+                    sendError : true
+                })
+            }
+        });
+    }
+
+    handleCloseSnackBar = () => {
+        this.setState({
+            sendSuccess : false,
+            sendError : false
+        });
+    };
+
+    handleCloseFeedBackBox = () => {
+        this.props.onSelectCancel();
+        this.setState({
+            charCtr : 0,
+            errorMsg: ''
+        });
+    };
+
+    render() {
+        const actions = [
+            <FlatButton
+                label="Cancel"
+                primary={true}

```

Comment:
open to discussion on this, but maybe make this button primary={false}, similar to programSelectionDialog.js:195 ?

Reply:
done